include/arch/arc/v2/irq.h

```diff
@@ -127,6 +127,12 @@ static ALWAYS_INLINE void z_arch_irq_unlock(unsigned int key)
 	__asm__ volatile("seti %0" : : "ir"(key) : "memory");
 }
 
+static ALWAYS_INLINE bool z_arch_irq_unlocked(unsigned int key)
+{
+	/* FIXME: need docs on key format! */
+	return false;
```

Comment:
key contains the relevant ARC status32 bits: interrupt enable (IE) and interrupt masking level.
The bit to test would be IE. This is bit 31 (0x8000_0000); 1=enabled; 0=disabled;